build.gradle

```diff
@@ -154,21 +154,33 @@ String checkForEapWithBuildNumberHigherThan(String latestEapBuildNumber) {
   return null
 }
 
-task updateEapBuildNumber {
-  doLast {
+class UpdateEapBuildNumber extends DefaultTask {
+
+  @Input
+  @Optional
+  @Option(option = 'exit-code', description = 'Option to return the exit code from task')
+  Boolean exitCode
+
+  @TaskAction
+  void execute() {
     Properties properties = project.intellijVersionsProperties()
     File propFile = project.intellijVersionsFile()
     String latestEapBuildNumber = project.intellijVersions.eapOfLatestSupportedMajor
     String newerEapBuildNumber = latestEapBuildNumber ? project.checkForEapWithBuildNumberHigherThan(latestEapBuildNumber.replace('-EAP-SNAPSHOT', '')) :
```

Comment:
While you're here... maybe extract the param of `project.checkForEapWithBuildNumberHigherThan(...)` to a separate var (`buildNumberThreshold`?), then `newerEapBuildNumber` can be just unconditionally set to

```groovy
   String newerEapBuildNumber = project.checkForEapWithBuildNumberHigherThan(buildNumberThreshold)
```